src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1529,6 +1530,30 @@ public function getContent($asResource = false)
         return $this->content;
     }
 
+    /**
+     * Return request body content as json decoded.
+     *
+     * @return mixed
+     *
+     * @throws JsonException
+     */
+    public function getJsonDecoded(int $depth = 512, int $options = 0)
+    {
+        $body = $this->getContent();
+
+        if ('' === $body) {
+            throw new JsonException('Unable to parse empty request body.');
+        }
+
+        $json = json_decode($body, true, $depth, $options);
```

Comment:
I would suggest to always set `JSON_THROW_ON_ERROR` and set `JSON_BIGINT_AS_STRING` by default. Losing precision by default doesn't look good to me.

Reply:
We cannot use `JSON_THROW_ON_ERROR` as long as we need to support php 7.1/7.2.

Reply:
we can with a `\defined(...)` call

Reply:
But then malformed JSON under php 7.3 would result in a different exception in than under 7.2. Why would we want that?

Reply:
Why so? We can throw a `JsonException` all the time.
https://github.com/symfony/polyfill/pull/167

Reply:
1) Yes, but right now, this PR invents its own `JsonException` that also implements `RequestExceptionInterface` which is treated differently by the framework than `\JsonException`.

2) If we always used `\JsonException`, we would gain nothing from conditionally adding `JSON_THROW_ON_ERROR` other than complexity. PHP is not faster in throwing exceptions than we are.